pkg/host/internal/sriov/sriov.go

```diff
@@ -588,15 +588,26 @@ func (s *sriov) GetNicSriovMode(pciAddress string) (string, error) {
 	log.Log.V(2).Info("GetNicSriovMode()", "device", pciAddress)
 
 	devLink, err := s.netlinkLib.DevLinkGetDeviceByName("pci", pciAddress)
-	if err != nil {
-		if errors.Is(err, syscall.ENODEV) {
-			// the device doesn't support devlink
-			return "", nil
-		}
+	if err != nil && !errors.Is(err, syscall.ENODEV) {
+		// the device doesn't support devlink
 		return "", err
 	}
-
-	return devLink.Attrs.Eswitch.Mode, nil
+	if devLink != nil && devLink.Attrs.Eswitch.Mode != "" {
+		return devLink.Attrs.Eswitch.Mode, nil
+	}
+	log.Log.V(2).Info("GetNicSriovMode(): can't get SRIOV mode with devlink, use sysfs method as fallback",
```

Comment:
do we really need this fallback? (i dont remember :) ) was this added to preserve functionality ?

ah maybe its for Intel NICs that support switchdev ?

Reply:
hm, I added this fallback because I was not able to query eswitch mode for NIC on my test env with inbox mlx driver on Ubuntu 22.04.  I tested again and it looks like the problem was with old FW. Now I'm able to eswitch mode after FW upgrade. I will revert this change.

Reply:
done, I removed fallback